modules/web/client/src/main/scala/observe/ui/components/queue/SessionQueue.scala

```diff
@@ -54,28 +64,27 @@ object SessionQueue:
       Css.Empty
 
   private def statusIconRenderer(
-    row:           SessionQueueRow,
-    selectedObsId: Option[Observation.Id]
+    loadingPotOpt: Option[Pot[Unit]],
+    statusOpt:     Option[SequenceState]
   ): VdomNode =
-    // <.div
-    val isFocused      = // row.active
-      selectedObsId.contains_(row.obsId)
-    // val selectedIconStyle = ObserveStyles.selectedIcon
     val icon: VdomNode =
-      row.status match
-        case SequenceState.Completed     => Icons.Check // clazz = selectedIconStyle)
-        case SequenceState.Running(_, _) => Icons.CircleNotch.withSpin(true)
-        //      loading = true,
+      (loadingPotOpt, statusOpt) match
+        case (Some(Pot.Pending), _)                                                        => Icons.CircleNotch.withSpin()
+        case (Some(Pot.Ready(_)), Some(SequenceState.Idle))                                => Icons.FileCheck
+        case (Some(Pot.Ready(_)), Some(SequenceState.Completed))                           =>
+          Icons.FileCheck // clazz = selectedIconStyle)
+        case (Some(Pot.Ready(_)), Some(SequenceState.Running(_, _)))                       => Icons.CircleNotch.withSpin()
         //      clazz = ObserveStyles.runningIcon
-        case SequenceState.Failed(_)     => EmptyVdom
+        case (Some(Pot.Ready(_)), Some(SequenceState.Failed(_))) | (Some(Pot.Error(_)), _) =>
+          Icons.FileCross
         // Icon(name = "attention", color = Red, clazz = selectedIconStyle)
         // case _ if b.state.rowLoading.exists(_ === index) =>
         // Spinning icon while loading
         // IconRefresh.copy(fitted = true, loading = true, clazz = ObserveStyles.runningIcon)
-        case _ if isFocused              =>             // EmptyVdom
-          Icons.CircleCheck.copy(size = IconSize.LG)
+        // case _ if isFocused              =>                 // EmptyVdom
```

Comment:
Commented code

Reply:
There's still seqexec code commented here which I'm keeping around until we finish porting the same functionality.